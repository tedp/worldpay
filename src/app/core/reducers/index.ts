import { ActionReducerMap, MetaReducer } from '@ngrx/store';
import { environment } from '../../../environments/environment';
import { PaymentsState, paymentReducer } from './payments.reducer';

export interface AppState {
  readonly payments: PaymentsState;
}

export const reducers: ActionReducerMap<AppState> = {
  payments: paymentReducer
};

export const metaReducers: MetaReducer<AppState>[] = !environment.production ? [] : [];

export * from './payments.reducer';

