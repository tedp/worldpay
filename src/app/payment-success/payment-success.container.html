<mat-card>
    <mat-card-header>
      <mat-card-title>Your payment is confirmed</mat-card-title>
    </mat-card-header>
</mat-card>  