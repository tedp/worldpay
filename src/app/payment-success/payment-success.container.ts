import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-payment-success',
  templateUrl: './payment-success.container.html',
  styleUrls: ['./payment-success.container.scss']
})
export class PaymentSuccessContainer implements OnInit {

  constructor() { }

  ngOnInit() {
  }

}
