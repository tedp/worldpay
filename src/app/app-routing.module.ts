import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import { TakePaymentContainer } from './take-payment/take-payment.container';
import { ConfirmPaymentContainer } from './confirm-payment/confirm-payment.container';
import { PaymentCancelledContainer } from './payment-cancelled/payment-cancelled.container';
import { PaymentSuccessContainer } from './payment-success/payment-success.container';

const routes: Routes = [
  { path: 'take-payment', component: TakePaymentContainer},
  { path: 'confirm-payment', component: ConfirmPaymentContainer},
  { path: 'payment-success', component: PaymentSuccessContainer},
  { path: 'payment-cancelled', component: PaymentCancelledContainer},
  { path: '',
    redirectTo: '/take-payment',
    pathMatch: 'full'
  },
 ];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule { }
