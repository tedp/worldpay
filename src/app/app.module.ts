import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { HttpClientModule } from '@angular/common/http';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { EffectsModule } from '@ngrx/effects';
import { PaymentsEffects } from './core/effects/payments.effects';
import { StoreModule } from '@ngrx/store';
import { reducers, metaReducers } from './core/reducers';
import { TakePaymentContainer } from './take-payment/take-payment.container';
import { ReactiveFormsModule } from '@angular/forms';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { MatInputModule, MatButtonModule, MatSelectModule, MatRadioModule, MatCardModule } from '@angular/material';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';
import { environment } from '../environments/environment';
import { ConfirmPaymentContainer } from './confirm-payment/confirm-payment.container';
import { PaymentSuccessContainer } from './payment-success/payment-success.container';
import { PaymentCancelledContainer } from './payment-cancelled/payment-cancelled.container';


@NgModule({
  declarations: [
    AppComponent,
    TakePaymentContainer,
    ConfirmPaymentContainer,
    PaymentSuccessContainer,
    PaymentCancelledContainer
  ],
  imports: [
    BrowserModule,
    HttpClientModule,
    AppRoutingModule,
    ReactiveFormsModule,
    EffectsModule.forRoot([PaymentsEffects]),
    StoreModule.forRoot(reducers, { metaReducers }),
    NoopAnimationsModule,
    MatInputModule,
    MatButtonModule,
    MatSelectModule,
    MatRadioModule,
    MatCardModule,
    StoreDevtoolsModule.instrument({ maxAge: 25, logOnly: environment.production })
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
