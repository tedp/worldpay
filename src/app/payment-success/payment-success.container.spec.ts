import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { PaymentSuccessContainer } from './payment-success.container';
import { AppModule } from '../app.module';
import { APP_BASE_HREF } from '@angular/common';

describe('PaymentSuccessComponent', () => {
  let component: PaymentSuccessContainer;
  let fixture: ComponentFixture<PaymentSuccessContainer>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [ AppModule ],
      providers:
      [
        {provide: APP_BASE_HREF, useValue : 'payment-success' }
      ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(PaymentSuccessContainer);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  // More test should be added here along the lines of take-payment-container.spec.ts

});
