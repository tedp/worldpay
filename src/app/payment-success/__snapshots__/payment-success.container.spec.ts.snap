// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentSuccessComponent should match snapshot 1`] = `
<app-payment-success>
  <mat-card
    class="mat-card"
  >
    <mat-card-header
      class="mat-card-header"
    >
      <div
        class="mat-card-header-text"
      >
        <mat-card-title
          class="mat-card-title"
        >
          Your payment is confirmed
        </mat-card-title>
      </div>
    </mat-card-header>
  </mat-card>
</app-payment-success>
`;
